src/Tribe/Admin_List.php

```diff
@@ -155,6 +158,41 @@ public static function sort_by_event_date( Array $clauses, WP_Query $wp_query )
 			return $clauses;
 		}
 
+		/**
+		 * Defines custom logic for filtering events table by aggregator record.
+		 *
+		 * @param array    $clauses    SQL clauses for fetching posts.
+		 * @param WP_Query $wp_query   A paginated query for items.
+		 *
+		 * @return array                  Modified SQL clauses.
+		 */
+		public static function filter_by_aggregator_record( array $clauses, WP_Query $wp_query ) {
+			// Check for event post type.
+			if ( $wp_query->get( 'post_type' ) !== Tribe__Events__Main::POSTTYPE ) {
+				return $clauses;
+			}
+
+			// Check if filtering by aggregator record.
+			$record_num = intval( tribe_get_request_var( 'aggregator_record', 0 ) );
+			if ( 0 === $record_num ) {
+				return $clauses;
+			}
+
+			global $wpdb;
+
+			// Add the record meta query if it is missing.
+			if ( ! preg_match( '/tribe_aggregator_record/', $clauses['join'] ) ) {
```

Comment:
```suggestion
			if ( ! preg_match( '/\\stribe_aggregator_record\\s/', $clauses['join'] ) ) {
```

`preg_match( '/tribe_aggregator_record/', $string )` will match `here_is_my_tribe_aggregator_record_table_alias` too; which is not correct; make sure there's a space before and after.

On a note: using `preg_match` with a simple string is not efficient, use `strpos` or `stripos`.